var/spack/repos/spack_repo/builtin/packages/paraview/package.py

```diff
@@ -232,22 +233,61 @@ class Paraview(CMakePackage, CudaPackage, ROCmPackage):
     depends_on("mpi", when="+mpi")
     conflicts("mpi", when="~mpi")
 
-    depends_on("qt@:4", when="@:5.2.0+qt")
-    depends_on("qt+sql", when="+qt")
-    with when("+qt"):
-        depends_on("qt+opengl", when="@5.3.0:+opengl2")
-        depends_on("qt~opengl", when="@5.3.0:~opengl2")
-
-    depends_on("gl@3.2:", when="+opengl2")
-    depends_on("gl@1.2:", when="~opengl2")
-    depends_on("glew")
-    depends_on("libxt", when="platform=linux ^[virtuals=gl] glx")
-
-    for plat in ["linux", "darwin", "freebsd"]:
+    # Handle X11 dependencies
+    # X is only used on Unix like platforms
+    conflicts("glx", when="~use_x")
+    # When on linux, X is required for Qt
+    for plat in ["linux", "freebsd"]:
         with when(f"platform={plat}"):
-            requires(
-                "^[virtuals=gl] glx", when="+qt", msg="Qt support requires GLX on non Windows"
-            )
+            requires("+use_x", when="+qt", msg="Qt support requires GLX on Linux/FreeBSD")
+
+    with when("+use_x"):
+        depends_on("libxt", when="@:5.12")
+        depends_on("libx11")
+        depends_on("libxcursor")
+        # When Qt and X are enabled, GLX is required in the runtime
+        requires("^[virtuals=gl] glx", when="@:5 +use_x")
+        depends_on("glx", when="@6: +use_x", type=("run"))
+
+    # ParaView@:5 support Qt5 and requires a GL provider to be known at
+    # build/link time.
+    with when("@:5"):
+        with when("+qt"):
+            depends_on("qt@:4", when="@:5.2.0")
+            depends_on("qt+sql")
+            depends_on("qt+opengl", when="@5.3.0:5 +opengl2")
+            depends_on("qt~opengl", when="@5.3.0:5 ~opengl2")
+            # Headless rendering not supported with Qt
+            conflicts("osmesa")
+            conflicts("egl")
+
+            # linux and freebsd require X11 when using Qt
+            for plat in ["linux", "freebsd"]:
+                with when(f"platform={plat}"):
+                    conflicts("~use_x", msg="X is required when building Qt on Linux")
```

Comment:
Isn't this redundant with L242:
```py
    for plat in ["linux", "freebsd"]:
        with when(f"platform={plat}"):
            requires("+use_x", when="+qt", msg="Qt support requires GLX on Linux/FreeBSD")
```

Reply:
Yes, it was moved at some point in my testing and got deleted from the v6 block but not the v5 block.